hkmc2/shared/src/test/mlscript/bbml/bbBasics.mls

```diff
@@ -149,7 +149,12 @@ fun foofoo(x) =
   if x is Int then
     let t = x + 1 in "foo"
 foofoo
-//│ Type: (¬Int ∨ Int) -> Str
+//│ Type: ['x, 'scrut, 'eff, 'cons] -> 'x ->{'eff} 'cons
+//│ Where:
+//│   'x <: ¬Int ∨ Int
+//│   'x#Int ∨  ∧ ⊥<:'eff ∧ 'scrut<:Int ∧ Int<:Int ∧ Str<:'cons
+//│   Int ∧ 'x <: 'scrut
+//│   'scrut#Int ∨  ∧ ⊥<:'eff ∧ 'scrut<:Int ∧ Int<:Int ∧ Str<:'cons
```

Comment:
For patmat without default cases, can't we just implicitly add a case that matches the negations of all previous patterns and constrains the scrutinee to be Bot?

Reply:
(This inferred type scheme, like the original one from BbML, misinterprets the patmat as being exhaustive, thus allowing `'x = ~Int` as a – spurious – solution.)